scripts/algcomptool/AlgCompTool.py

```diff
@@ -0,0 +1,335 @@
+			alg2CompSegment = alg2[alg2['sub'] == subject]
+			alg2Sample = 0
+			while alg2CompSegment.iloc[alg2Sample,classColLocation] != 1:
+				alg2Sample = alg2Sample + 1
+
+			if alg1Sample > alg2Sample:
+				fileComp.write("Alg1 < Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+			elif alg1Sample < alg2Sample:
+				fileComp.write("Alg1 > Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+			elif alg1Sample == alg2Sample:
+				#Same prediction level, so must compare ratios between highest weight prediction and this wrong preiction
+				alg1Ratio = alg1CompSegment.iloc[0,scoreColLocation]/alg1CompSegment.iloc[alg1Sample,scoreColLocation]
+				alg2Ratio = alg2CompSegment.iloc[0,scoreColLocation]/alg2CompSegment.iloc[alg2Sample,scoreColLocation]
+				#The bigger the ratio the worse the prediction: i.e. 0.5 (first), 0.3, 0.001, 0.0001 (our prediction) vs.
+				#0.5 (first), 0.2 (our prediction)
+				if alg1Ratio > alg2Ratio:
+					fileComp.write("Alg1 < Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+				elif alg1Ratio < alg2Ratio:
+					fileComp.write("Alg1 > Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+		fileComp.close
+
+		try:
+			fileConfAlg1 = open(alg1confusion,"w")
+		except:
+			print("Error writing to the specified alg1confusion file, please check -oa1 argument", file=sys.stderr)
+			sys.exit(1)
+		
+		fileConfAlg1.write("sid,sub,oid,obj,conf\n")
+
+		for sample in range(0, alg1.shape[0]):
+			if sample % alg1DistinctObjNum == 0:
+				if alg1.iloc[sample, classColLocation] == 1:
+					fileConfAlg1.write(str(alg1.iloc[sample, idColLocation]) + "," + str(alg1.iloc[sample, subjectColLocation].replace("dbr:","")) + "," + str(alg1.iloc[sample, objectIDColLocation]) + "," + str(alg1.iloc[sample, objectColLocation].replace("dbr:","")) + ",TP" + "\n")
+				else:
+					fileConfAlg1.write(str(alg1.iloc[sample, idColLocation]) + "," + str(alg1.iloc[sample, subjectColLocation].replace("dbr:","")) + "," + str(alg1.iloc[sample, objectIDColLocation]) + "," + str(alg1.iloc[sample, objectColLocation].replace("dbr:","")) + ",FP" + "\n")
+			else:
+				if alg1.iloc[sample, classColLocation] == 1:
+					fileConfAlg1.write(str(alg1.iloc[sample, idColLocation]) + "," + str(alg1.iloc[sample, subjectColLocation].replace("dbr:","")) + "," + str(alg1.iloc[sample, objectIDColLocation]) + "," + str(alg1.iloc[sample, objectColLocation].replace("dbr:","")) + ",FN" + "\n")
+				else:
+					fileConfAlg1.write(str(alg1.iloc[sample, idColLocation]) + "," + str(alg1.iloc[sample, subjectColLocation].replace("dbr:","")) + "," + str(alg1.iloc[sample, objectIDColLocation]) + "," + str(alg1.iloc[sample, objectColLocation].replace("dbr:","")) + ",TN" + "\n")
+
+		fileConfAlg1.close
```

Comment:
Changed the fileConfAlg1.write and fileConfAlg2.write procedures to call a function which can write both. In many cases it is possible to include functions; however, we are only repeating an event once or twice, and in my opinion there is no need to further abstract code if we call the block only a few times (i.e. 1-2) but if we call code 3+ times then yes it's a good idea to write a function. Let me know if you agree with this.